core/src/main/java/org/opensearch/sql/expression/window/patterns/BufferPatternWindowFunction.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.expression.window.patterns;
+
+import static org.opensearch.sql.utils.ExpressionUtils.format;
+
+import java.util.List;
+import java.util.Locale;
+import lombok.EqualsAndHashCode;
+import org.opensearch.sql.common.patterns.BrainLogParser;
+import org.opensearch.sql.data.model.ExprStringValue;
+import org.opensearch.sql.data.model.ExprValue;
+import org.opensearch.sql.data.type.ExprCoreType;
+import org.opensearch.sql.data.type.ExprType;
+import org.opensearch.sql.expression.Expression;
+import org.opensearch.sql.expression.FunctionExpression;
+import org.opensearch.sql.expression.NamedArgumentExpression;
+import org.opensearch.sql.expression.env.Environment;
+import org.opensearch.sql.expression.function.BuiltinFunctionName;
+import org.opensearch.sql.expression.window.WindowDefinition;
+import org.opensearch.sql.expression.window.WindowFunctionExpression;
+import org.opensearch.sql.expression.window.frame.BufferPatternRowsWindowFrame;
+import org.opensearch.sql.expression.window.frame.WindowFrame;
+
+@EqualsAndHashCode(callSuper = true)
+public class BufferPatternWindowFunction extends FunctionExpression
```

Comment:
`order by` expression in window definition decides who're peers right? What's the order by expression for buffer offline pattern function? I didn't find it and may miss it in design doc:

```
Project[message#1, patterns_field#2]
+- Window[brain(message#1), windowsSpec(partitionBy=null, frame=PeerRowsWindowFrame)]
   +- OpenSearchIndexScan
```